core/src/main/scala/kafka/log/Log.scala

```diff
@@ -918,6 +954,7 @@ class Log(@volatile var dir: File,
                                 preallocate = config.preallocate))
       updateLogEndOffset(newOffset)
       this.recoveryPoint = math.min(newOffset, this.recoveryPoint)
+      this.logStartOffset = math.min(newOffset, this.logStartOffset)
```

Comment:
If we truncate log using `truncateFullyAndStartAt` to an offset that is higher than the `logStartOffset`, I think we should keep the logStartOffset not changed instead of increasing it, right? It seems similar to the existing logic of updating `recoveryPoint`, where we use `math.min`. Another argument is that it is safer to assume a small logStartOffset.